datastore.go

```diff
@@ -52,7 +59,9 @@ var DefaultOptions Options
 
 func init() {
 	DefaultOptions = Options{
-		gcDiscardRatio: 0.1,
+		GcDiscardRatio: 0.5,
+		MaxGcDuration:  1 * time.Minute,
+		GcInterval:     15 * time.Minute,
```

Comment:
Allow to disable and leave disabled by default. It is pretty bad that badger is going to go to a crawl periodically by default when in many cases there might be no need at all to GC (imho).

Reply:
@hsanjuan  But if the client never explicitly calls `CollectGarbage`, we will never delete anything which was the motivation for [this issue](https://github.com/ipfs/go-ds-badger/issues/51). If frequent calls to Badger GC is a concern, I think we should change the interval to a higher value of say 1 hour ?

Reply:
@hsanjuan specifically, this is an issue for things like the DHT and storing the peerstore on disk. As-is, badger will _never_ delete these records, even if tell it to.

Reply:
I think it's wrong this has been enabled by default (and silently to any end-user). I understand you need periodic GC in DHT etc., but this is a wrapper on badger. You are imposing non-default badger behaviour, that didn't exist before, and that cannot be disabled (unless setting the gc period to max and keeping an extra goroutine ad-infinitum).

We use badger to store crdt, append-only data that does not need to be GC'ed automatically. If you need automatic GC anywhere, add the feature and enable there.

And honestly, using badger to store a peerstore is probably overkill. Badger is an append-only store. If you write/delete so often that this causes a log-size issue then badger is probably not the best choice and the size of the peerstore does not seem to grow so big that would warrant the use of badger.

Reply:
Having a datastore that doesn't delete anything when requested is incorrect and unexpected. The user shouldn't have to "turn on delete".

1. We can definitely make it possible to turn this off (by setting a 0 interval).
2. If nothing has been deleted, badger will sample a _single_ log file every time period. If there was nothing to delete, we'll wait another GC period (15m).

> And honestly, using badger to store a peerstore is probably overkill. Badger is an append-only store. If you write/delete so often that this causes a log-size issue then badger is probably not the best choice and the size of the peerstore does not seem to grow so big that would warrant the use of badger.

Badger is _not_ an append-only store. It's designed to be periodically garbage collected.

Reply:
I've filed an issue as garbage collecting really should be a no-op when there's nothing to delete: https://github.com/dgraph-io/badger/issues/1079

Reply:
Ok, so, I have answer: Worst case scenario, if you've never deleted something, the badger datastore will sample a _single_ 64MiB value log every 15 minutes.